clang/lib/Driver/Driver.cpp

```diff
@@ -4363,20 +4363,34 @@ class OffloadingActionBuilder final {
         ActionList WrapperInputs;
         // post link is not optional - even if not splitting, always need to
         // process specialization constants
-        types::ID PostLinkOutType =
-            isNVPTX ? types::TY_LLVM_BC : types::TY_Tempfiletable;
         auto *PostLinkAction = C.MakeAction<SYCLPostLinkJobAction>(
-            FullDeviceLinkAction, PostLinkOutType);
+            FullDeviceLinkAction, types::TY_Tempfiletable);
         PostLinkAction->setRTSetsSpecConstants(!isAOT);
 
+        constexpr char COL_CODE[] = "Code";
+
         if (isNVPTX) {
-          Action *FinAction =
-              finalizeNVPTXDependences(PostLinkAction, (*TC)->getTriple());
-          WrapperInputs.push_back(FinAction);
+          // Make extraction copy the only remaining code file instead of
+          // creating a new table with a single entry.
+          // TODO: Process all PTX code files in file table when code-splitting
+          //       is implemented to PTX target.
```

Comment:
Actually, this might be the only thing which prevents PTX to work with code splitting, where PTX flow is unable to process a file list :) So, I'd suggest to change:
```suggestion
          // TODO: Process all PTX code files in file table to enable code splitting for PTX target.
```

Reply:
Oh, maybe so! I have changed the TODO as you suggested.